src/handlers/session.ts

```diff
@@ -1,14 +1,14 @@
-import { IncomingMessage } from 'http';
+import { NextApiResponse, NextApiRequest } from 'next';
 
 import { ISession } from '../session/session';
 import { ISessionStore } from '../session/store';
 
 export default function sessionHandler(sessionStore: ISessionStore) {
-  return (req: IncomingMessage): Promise<ISession | null | undefined> => {
+  return (req: NextApiRequest, res: NextApiResponse): Promise<ISession | null | undefined> => {
```

Comment:
This is a breaking change. Can we make `res` optional somehow?

Reply:
`CookieSessionStore` calls save which seems to require it.  I am not sure we can avoid that breaking change.